omx/gstomxvideodec.c

```diff
@@ -2771,3 +2798,62 @@ gst_omx_video_dec_decide_allocation (GstVideoDecoder * bdec, GstQuery * query)
 
   return TRUE;
 }
+
+static gboolean
+gst_omx_video_dec_sink_event (GstVideoDecoder * decoder, GstEvent * event)
+{
+  GstOMXVideoDec *self = GST_OMX_VIDEO_DEC (decoder);
+  gboolean ret = TRUE;
+
+  switch (GST_EVENT_TYPE (event)) {
+    case GST_EVENT_FLUSH_START:
+      GST_VIDEO_DECODER_STREAM_LOCK (self);
+      self->flush_flag = TRUE;
+      self->started = FALSE;
+      self->downstream_flow_ret = GST_FLOW_OK;
+      GST_VIDEO_DECODER_STREAM_UNLOCK (self);
+      /* Stop srcpad task */
+      gst_pad_stop_task (GST_VIDEO_DECODER_SRC_PAD (self));
+      /* Flush omx ports */
+      gst_omx_port_set_flushing (self->dec_in_port, 5 * GST_SECOND, TRUE);
+      gst_omx_port_set_flushing (self->dec_out_port, 5 * GST_SECOND, TRUE);
+      break;
+    case GST_EVENT_FLUSH_STOP:
+      /* Unflush omx ports */
+      gst_omx_port_set_flushing (self->dec_in_port, 5 * GST_SECOND, FALSE);
+      gst_omx_port_set_flushing (self->dec_out_port, 5 * GST_SECOND, FALSE);
+      GST_VIDEO_DECODER_STREAM_LOCK (self);
+      self->flush_flag = FALSE;
+      self->downstream_flow_ret = GST_FLOW_OK;
+      GST_VIDEO_DECODER_STREAM_UNLOCK (self);
+      break;
+    default:
+      break;
+  }
+
+  ret =
+      GST_VIDEO_DECODER_CLASS (gst_omx_video_dec_parent_class)->sink_event
+      (decoder, event);
+
+  if (!ret) {
+    GST_ERROR_OBJECT (self, "Error processing sink event");
+    goto exit;
+  }
+
+  switch (GST_EVENT_TYPE (event)) {
+    case GST_EVENT_SEGMENT:
```

Comment:
I think we should let the default function handle this as well. Is there a reason for not doing so?

Reply:
All events are handled by the gstvideodecoder parent class, but some of them are handled also by the omxvideodec class. Flush events are handled by omxvideodec (child) before the parent, while the segment event is handled by omxvideodec after the parent (line 2835). This is done this way because is better that the parent class refresh all the segment related attributes before omxvideodec restores the loop task.